src/render_tree.rs

```diff
@@ -0,0 +1,563 @@
+            position: Position::new_from_existing(position),
+        };
+
+        position.offset_y(font_size);
+        position.offset_y(margin);
+
+        new_node
+    }
+
+    pub fn new_heading3(position: &mut Position) -> Self {
+        let margin = 9.36;
+        position.offset_y(margin);
+
+        let new_heading = TextNode::new_heading3();
+        let font_size = new_heading.font_size;
+
+        let new_node = Node {
+            node_type: NodeType::Text(new_heading),
+            margin: Rectangle::with_values(margin, 0., 0., margin),
+            padding: Rectangle::new(),
+            parent: None,
+            next_sibling: None,
+            children: Vec::new(),
+            position: Position::new_from_existing(position),
+        };
+
+        position.offset_y(font_size);
+        position.offset_y(margin);
+
+        new_node
+    }
+
+    pub fn new_heading4(position: &mut Position) -> Self {
+        let margin = 10.64;
+        position.offset_y(margin);
+
+        let new_heading = TextNode::new_heading4();
+        let font_size = new_heading.font_size;
+
+        let new_node = Node {
+            node_type: NodeType::Text(new_heading),
+            margin: Rectangle::with_values(margin, 0., 0., margin),
+            padding: Rectangle::new(),
+            parent: None,
+            next_sibling: None,
+            children: Vec::new(),
+            position: Position::new_from_existing(position),
+        };
+
+        position.offset_y(font_size);
+        position.offset_y(margin);
+
+        new_node
+    }
+
+    pub fn new_heading5(position: &mut Position) -> Self {
+        let margin = 11.089;
+        position.offset_y(margin);
+
+        let new_heading = TextNode::new_heading5();
+        let font_size = new_heading.font_size;
+
+        let new_node = Node {
+            node_type: NodeType::Text(new_heading),
+            margin: Rectangle::with_values(margin, 0., 0., margin),
+            padding: Rectangle::new(),
+            parent: None,
+            next_sibling: None,
+            children: Vec::new(),
+            position: Position::new_from_existing(position),
+        };
+
+        position.offset_y(font_size);
+        position.offset_y(margin);
+
+        new_node
+    }
+
+    pub fn new_heading6(position: &mut Position) -> Self {
+        let margin = 12.489;
+        position.offset_y(margin);
+
+        let new_heading = TextNode::new_heading6();
+        let font_size = new_heading.font_size;
+
+        let new_node = Node {
+            node_type: NodeType::Text(new_heading),
+            margin: Rectangle::with_values(margin, 0., 0., margin),
+            padding: Rectangle::new(),
+            parent: None,
+            next_sibling: None,
+            children: Vec::new(),
+            position: Position::new_from_existing(position),
+        };
+
+        position.offset_y(font_size);
+        position.offset_y(margin);
+
+        new_node
+    }
```

Comment:
I think, here it is also possible to get rid of huge chunks of the code because it's repeating very much. I can also commit something for this here if you want.

Reply:
You can put this under the same branch mentioned above